### PR TITLE
Update for project level nullable setting

### PR DESCRIPTION
**Pull request by BillWagner**
And address feedback missed on the last PR for nullables.

/cc @pkulikov

### CODE REVIEW

File: docs/csharp/nullable-references.md [hunk 1]
```diff
@@ -57,21 +57,21 @@ The behavior is the same as previous versions of C#.
 
 The compiler uses the following rules in an enabled nullable context:
 
-- Any variable of a reference type is a **non-nullable reference type**.
+- Any variable of a reference type is a **non-nullable reference**.
 - Any non-nullable reference may be dereferenced safely.
-- Any nullable reference type (noted by `?` after the type in the variable declaration) may be null, and must be verified as not null before it can be dereferenced.
+- Any nullable reference type (noted by `?` after the type in the variable declaration) may be null. Static analysis determines if it the value is known to be not null when it is dereferenced. If not, the compiler warns you.
```
Comment (pkulikov) [inline docs/csharp/nullable-references.md]: There's some grammar hiccup in the "Static analysis determines if it the value is known..." sentence. Should "it" be removed?

File: docs/csharp/nullable-references.md [hunk 2]
```diff
@@ -57,21 +57,21 @@ The behavior is the same as previous versions of C#.
 
 The compiler uses the following rules in an enabled nullable context:
 
-- Any variable of a reference type is a **non-nullable reference type**.
+- Any variable of a reference type is a **non-nullable reference**.
 - Any non-nullable reference may be dereferenced safely.
-- Any nullable reference type (noted by `?` after the type in the variable declaration) may be null, and must be verified as not null before it can be dereferenced.
+- Any nullable reference type (noted by `?` after the type in the variable declaration) may be null. Static analysis determines if the value is known to be not null when it is dereferenced. If not, the compiler warns you.
```
Comment (rpetrusha) [inline docs/csharp/nullable-references.md]: nit: is non-null preferable to not null?

File: docs/csharp/nullable-references.md [hunk 3]
```diff
@@ -57,21 +57,21 @@ The behavior is the same as previous versions of C#.
...
 
 1. The variable has been definitely assigned to a non-null value.
 1. The variable has been checked against null before de-referencing it.
 
-The compiler enforces that non-nullable references may never be set to the null value. You must initialize them to a non-null value. If you don't, the compiler warns that a non-nullable reference wasn't initialized. The compiler also warns you whenever you assign a non-nullable reference to a value that **maybe null**. That implies you can only assign a non-nullable reference to a nullable reference if that nullable reference is **not null**.
+The compiler enforces that non-nullable references may never be set to the null value. You must initialize them to a non-null value. If you don't, the compiler warns that a non-nullable reference wasn't initialized. The compiler also warns you whenever you assign a non-nullable reference to a nullable reference that **maybe null**. That implies you can only assign a non-nullable reference to a nullable reference if that nullable reference is **not null**.
```
Comment (rpetrusha) [inline docs/csharp/nullable-references.md]: maybe null --> may be null

**Review by rpetrusha — APPROVED** [review]
LGTM, @BillWagner. I've left a couple of minor nits for you to consider. You can merge when you're ready.

**Pull request merged**